### ISSUE TITLE
Web-Assembly samples - .png / .jpeg for textures?

### ISSUE THREAD

**Issue by sarathsi** [google/filament#440]
Referring to Web-Assembly samples, I see only .ktx format is used for textures.
(https://github.com/google/filament/blob/9e5f44bd3458bbf92ae3aa7cd733a5de359d7d6e/samples/web/suzanne.html)

I tried to use .jpeg, .png as textures by modifying samples/web/suzanne.cpp to use  getRawFile()
    static auto metallic = filaweb::getRawFile("metallic");
    static auto roughness = filaweb::getRawFile("roughness");
    static auto normal = filaweb::getRawFile("normal");

But it did not work. Is it supposed to work or  not supported?

**Comment by prideout** [comment]
Hi, our new JavaScript API supports PNG. For example the [test_parquet](https://github.com/google/filament/blob/master/libs/filamentjs/tests/test_parquet.js) sample calls `engine.createTextureFromPng`.

If you need to use C++, then it's a bit more involved because you would need to include a PNG decoder (e.g. stb) into your wasm module.

**Comment by sarathsi** [comment]
Hi, Thanks for info. 
About using JS API, should we use filament.js generated in  filament/out/cmake-webgl-release/libs/filamentjs/ when building for webgl release ? 

gl-matrix-min.js: Is it version 2.4.0? from https://gist.github.com/pbfy0/3a1717978fc68adaa7a878a2089db731 ? or any other version?

**Comment by prideout** [comment]
Correct.  I recommend following the tutorial found in the docs folder in the [web release](https://filament-build.storage.googleapis.com/badges/build_link_web.html).  I'll fix up the root README to make this more clear.

For matrices and vectors, the JS API simply takes small JS arrays; usage of glMatrix is not required, but it is recommended.  The tutorials use glMatrix 2.8.1 but I think 2.0 or higher is fine.
